subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/DependencyManagementResultsAsInputsIntegrationTest.groovy

```diff
@@ -0,0 +1,391 @@
+ * limitations under the License.
+ */
+
+package org.gradle.integtests.resolve
+
+import groovy.test.NotYetImplemented
+import org.gradle.api.internal.artifacts.DefaultModuleIdentifier
+import org.gradle.api.internal.artifacts.DefaultModuleVersionIdentifier
+import org.gradle.api.internal.artifacts.result.DefaultResolvedVariantResult
+import org.gradle.api.internal.attributes.ImmutableAttributesFactory
+import org.gradle.integtests.fixtures.AbstractHttpDependencyResolutionTest
+import org.gradle.integtests.fixtures.daemon.DaemonLogsAnalyzer
+import org.gradle.internal.Describables
+import org.gradle.internal.component.external.model.DefaultModuleComponentArtifactIdentifier
+import org.gradle.internal.component.external.model.DefaultModuleComponentIdentifier
+import org.gradle.internal.component.external.model.ImmutableCapability
+import org.gradle.test.fixtures.file.TestFile
+import spock.lang.Issue
+
+class DependencyManagementResultsAsInputsIntegrationTest extends AbstractHttpDependencyResolutionTest {
+
+    def setup() {
+        settingsFile << """
+            includeBuild 'composite-lib'
+            rootProject.name = 'root'
+            include 'project-lib'
+        """
+        file('composite-lib/settings.gradle') << ""
+        file('composite-lib/build.gradle') << """
+            plugins { id 'java' }
+            group = 'composite-lib'
+        """
+        mavenRepo.module("org.external", "external-lib").publish()
+        mavenRepo.module("org.external", "external-tool").publish()
+        file('lib/file-lib.jar') << 'content'
+        buildFile << """
+            project(':project-lib') { apply plugin: 'java' }
+            configurations { compile }
+            repositories { maven { url "${mavenRepo.uri}" } }
+            dependencies {
+                compile 'org.external:external-lib:1.0'
+                compile project('project-lib')
+                compile files('lib/file-lib.jar')
+                compile 'composite-lib:composite-lib'
+            }
+        """
+        withOriginalSourceIn("project-lib")
+        withOriginalSourceIn("composite-lib")
+        // We need fresh daemons to exercise loading snapshots from disk in tests below
+        executer.requireIsolatedDaemons()
+    }
+
+    def "can use #type as task input"() {
+        given:
+        buildFile << """
+            import ${DefaultModuleIdentifier.name}
+            import ${DefaultModuleVersionIdentifier.name}
+            import ${DefaultModuleComponentIdentifier.name}
+            import ${ImmutableCapability.name}
+            import ${DefaultModuleComponentArtifactIdentifier.name}
+            import ${ImmutableAttributesFactory.name}
+            import ${DefaultResolvedVariantResult.name}
+            import ${Describables.name}
+
+            abstract class TaskWithInput extends DefaultTask {
+
+                @Input
+                abstract Property<$type> getInput()
```

Comment:
🟡 Are `SetProperty<$type>` and `ListProperty<$type>` input properties also supported? Is it redundant to test them here?

Reply:
They are supported and yes it would be redundant to test them here.